prov/efa/src/efa_cntr.c

```diff
@@ -77,30 +77,80 @@ static int efa_cntr_wait(struct fid_cntr *cntr_fid, uint64_t threshold, int time
 static void efa_cntr_progress(struct util_cntr *cntr)
 {
 	struct util_srx_ctx *srx_ctx;
+	struct efa_cntr *efa_cntr;
+
 	srx_ctx = cntr->domain->srx->ep_fid.fid.context;
+	efa_cntr = container_of(cntr, struct efa_cntr, util_cntr);
 
 	ofi_genlock_lock(srx_ctx->lock);
+	if (efa_cntr->shm_cntr)
+		fi_cntr_read(efa_cntr->shm_cntr);
 	ofi_cntr_progress(cntr);
 	ofi_genlock_unlock(srx_ctx->lock);
 }
 
+static int efa_cntr_close(struct fid *fid)
+{
+	struct efa_cntr *cntr;
+	int ret, retv;
+
+	retv = 0;
+	cntr = container_of(fid, struct efa_cntr, util_cntr.cntr_fid.fid);
+
+	if (cntr->shm_cntr) {
```

Comment:
Should we do NULL cntr check here?

Reply:
it's a container of, it's unlikely that cntr can be NULL here.